server-common/src/main/java/org/apache/kafka/server/config/ServerQuotaConfigs.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.config;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ServerQuotaConfigs {
```

Comment:
1. We have 2 `QuotaConfigs` one in `client` and another in `DynamicConfig` and I was trying to avoid dealing with a third one in the code. 
2. This config are also used for replication and controller mutations so it's not only client.

I can call it `CommonQuotaConfigs` otherwise I can name it `QuotaConfigs` but will need to make sure we don't don't get into import ambiguity by importing each config prop or default value separately.

Reply:
There is a `QuotaConfigs` in client module already. That could be a bit confused if we put a `QuotaConfigs` in the server-common module.

Reply:
Am not attached to the name but let me know which would make more sense or if you have any other suggestions/approach

Reply:
`org.apache.kafka.common.config.internals.QuotaConfigs` does not seem to be used by clients at all. So I wonder if we could move and merge it with this one.

Reply:
> org.apache.kafka.common.config.internals.QuotaConfigs does not seem to be used by clients at all. So I wonder if we could move and merge it with this one.

that is a good suggestion. `org.apache.kafka.common.config.internals.QuotaConfigs` is used by `core` and `metadata` module, so merging it to `server-common` makes sense to me.

Reply:
this is a good point. Let me see if I can move it to `server-common`. It seem to be small class anyway. We still will have another one in DynamicConfig but this one only have one method to decide `isClientOrUserQuotaConfig` by calling `org.apache.kafka.common.config.internals.QuotaConfigs.isClientOrUserConfig` so maybe this also can be merged